src/nxt_http_request.c

```diff
@@ -957,6 +957,11 @@ nxt_http_arguments_parse(nxt_http_request_t *r)
 
     r->args_decoded.start = dst_start;
 
+    if (nxt_slow_path(r->args->length == 0)) {
+        r->args_decoded.length = 0;
+        goto end;
+    }
+
     start = r->args->start;
     end = start + r->args->length;
 
```

Comment:
Again, if this is because _r->args->start_ can be NULL, then I'd check for that.

Reply:
This results in less code:

```c
start = r->args->start ?: (u_char) "";
```

Reply:
We'll short circuit out the for () loop the only question is do we want to (or not) do the bit of code between the for () loop and the end label in such situations?

Reply:
Short circuiting optimizes for the case where .length == 0, at the cost of adding some more code (possibly pessimizing the rest of the cases by one op or so).  I guess it depends on how likely each of the cases are.

Reply:
Sorry I wasn't clear, the short circuiting is fine. But by not just going to `end:` when ->start is NULL means we'll execute some other stuff that we _may_ not want to...

Looking closer, assuming the `dst = dst_start` in the for loop initialiser happens, then we will execute

```c
r->args_decoded.length = dst - r->args_decoded.start;
```

Which will equate to `dst - dst` which should be `0` and

```c
    if (name_length != 0 || dst != dst_start) {
```

wil equate to false. So I think those are OK.

So yes, the ternary should also do the trick...

Reply:
> Again, if this is because _r->args->start_ can be NULL, then I'd check for that.

I can.

For example:
```
/foo: r->args->start is NULL
/foo?: r->args->start is not NULL
```

And for the configuration:
```
{
    "match": {
          "query": ""
    }
}
```
The pattern related to `query` is also NULL.
```
pattern->start = NULL;
```

So it looks like we don't need to touch `r->args_decoded` if there is no UB reported. Otherwise, not sure if we'll have to rethink the parrten->start to query "" value.

I'd would suggest fixing the case of `r->args_start` is NULL. For example:
```
if (nxt_slow_path(r->args->start == NULL)) {
    goto end;
}
```

Reply:
This reminds me of <https://github.com/mkirchner/linked-list-good-taste>.

Reply:
Heh, I actually have that email from Linus saved off for whenever I need to refer to it...

I think what Alex is saying is that with his ternary suggestion the code just flows through and does the right thing ragardless if `r->args->start` is `NULL` or not.

Reply:
I'd go with Alex's suggestion and fix it like you did in `NJS: avoiding arithmetic ops with NULL pointer in r->args`